modules/fx/arrow-fx/src/main/kotlin/arrow/fx/IO.kt

```diff
@@ -785,7 +784,7 @@ sealed class IO<out E, out A> : IOOf<E, A> {
     // Errors short-circuit
     override fun <B> map(f: (Nothing) -> B): IO<E, B> = this
 
-    override fun unsafeRunTimedTotal(limit: Duration): Option<Nothing> = None
+    override fun unsafeRunTimedTotal(limit: Duration): Option<Either<E, Nothing>> = Some(Left(error))
```

Comment:
Are we sure? `cats-bio` [throws an exception](https://github.com/LukaJCB/cats-bio/blob/2bcf0a496838f41c94683abd1d524876606bd17a/src/main/scala/cats/effect/bio/BIO.scala#L324( containing the error.toString in this case.

Reply:
Why would you break type-safety and throw? If the user wants to do so he can still throw. That would also mean that `IO<E, A>.unsafeRunSync` returns `A` instead of `Either<E, A>`.

Reply:
I guess I imagined Haskell. I'm not sure how it is because no experience, but I imagined the language and paradigm would push you to control all types of errors within your program, and otherwise assume they're unexpected and throw (logs error to repl when running the IO). We are kind of in a limbo in Kotlin since our program is still running outside of the pure boundaries and you can still add more logics on top of it.